pkg/git/libgit2/managed/http.go

```diff
@@ -209,6 +195,20 @@ func createClientRequest(targetUrl string, action git2go.SmartServiceAction, t *
 		return nil, nil, err
 	}
 
+	// Add any provided certificate to the http transport.
+	if opts.AuthOpts != nil {
+		req.SetBasicAuth(opts.AuthOpts.Username, opts.AuthOpts.Password)
```

Comment:
This may lead to the previous "empty credentials" issue for public repositories which resulted 403 in GitLab. We could change GitKit to refuse empty UserName/Password to test this behaviour.